src/metabase/lib/schema/validate.cljc

```diff
@@ -0,0 +1,38 @@
+(ns metabase.lib.schema.validate
+  (:require
+   [metabase.lib.schema.common :as common]
+   [metabase.util.malli.registry :as mr]))
+
+(mr/def ::missing-column
+  [:map
+   [:type [:= :validate/missing-column]]
+   [:name :string]])
+
+(mr/def ::missing-table-alias
+  [:map
+   [:type [:= :validate/missing-table-alias]]
+   [:name :string]])
+
+(mr/def ::duplicate-column
+  [:map
+   [:type [:= :validate/duplicate-column]]
+   [:name :string]])
+
+(mr/def ::syntax-error
+  [:map
+   [:type [:= :validate/syntax-error]]])
+
+(mr/def ::validation-error
```

Comment:
At the end of the day aren't these *all* validation errors? Maybe this name could be a little clearer. (Not sure what this is actually for... `::schema-validation-error` maybe?)